dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/validation/OAuthFlowsValidator.java

```diff
@@ -34,6 +37,40 @@ private OAuthFlowsValidator() {}
     /** {@inheritDoc} */
     @Override
     public void validate(ValidationHelper helper, Context context, String key, OAuthFlows t) {
-        // TODO Auto-generated method stub
+        if (t.getImplicit() != null) {
+            OAuthFlow implicit = t.getImplicit();
+            String type = "implicit OAuthFlow";
+            if (StringUtils.isNotBlank(implicit.getTokenUrl())) {
+                final String message = Tr.formatMessage(tc, "nonApplicableField", "tokenUrl", implicit.getTokenUrl(), type);
+                helper.addValidationEvent(new ValidationEvent(ValidationEvent.Severity.WARNING, null, message));
+            }
+            ValidatorUtils.validateRequiredField(implicit.getAuthorizationUrl(), context, "authorizationUrl").ifPresent(helper::addValidationEvent);
+        }
+        if (t.getPassword() != null) {
+            OAuthFlow password = t.getPassword();
+            String type = "password OAuthFlow";
+            if (StringUtils.isNotBlank(password.getAuthorizationUrl())) {
+                final String message = Tr.formatMessage(tc, "nonApplicableField", "authorizationUrl", password.getAuthorizationUrl(), type);
+                helper.addValidationEvent(new ValidationEvent(ValidationEvent.Severity.WARNING, null, message));
+            }
+            ValidatorUtils.validateRequiredField(password.getTokenUrl(), context, "tokenUrl").ifPresent(helper::addValidationEvent);
+        }
+        if (t.getClientCredentials() != null) {
+            OAuthFlow clientCred = t.getClientCredentials();
+            String type = "clientCredentials OAuthFlow";
+            if (StringUtils.isNotBlank(clientCred.getAuthorizationUrl())) {
+                final String message = Tr.formatMessage(tc, "nonApplicableField", "authorizationUrl", clientCred.getAuthorizationUrl(), type);
+                helper.addValidationEvent(new ValidationEvent(ValidationEvent.Severity.WARNING, null, message));
+            }
+            ValidatorUtils.validateRequiredField(clientCred.getTokenUrl(), context, "tokenUrl").ifPresent(helper::addValidationEvent);
+        }
+        if (t.getAuthorizationCode() != null) {
+            OAuthFlow authCode = t.getAuthorizationCode();
+            if (StringUtils.isBlank(authCode.getTokenUrl())) {
```

Comment:
Wondering if this check is required - helper method will handle null/empty.

Reply:
You are right, removed it.